src/net/smooth/parser.js

```diff
@@ -245,18 +262,10 @@ function createSmoothStreamingParser(parserOptions={}) {
     assert(representations.length, "parser: adaptation should have at least one representation");
 
     // apply default codec if non-supported
-    let codecs = representations[0].codecs;
-    if (!codecs) {
-      codecs = DEFAULT_CODECS[type];
-      representations.forEach((rep) => rep.codecs = codecs);
-    }
+    representations.forEach((rep) => rep.codecs = rep.codecs || DEFAULT_CODECS[type]);
 
     // apply default mimetype if non-supported
-    let mimeType = representations[0].mimeType;
-    if (!mimeType) {
-      mimeType = DEFAULT_MIME_TYPES[type];
-      representations.forEach((rep) => rep.mimeType = mimeType);
-    }
+    representations.forEach((rep) => rep.mimeType = rep.mimeType || DEFAULT_MIME_TYPES[type]);
```

Comment:
I saw that all codecs/mimeTypes were set to default when first representation had them empty